project-management-infrastructure/src/main/java/life/qbic/projectmanagement/infrastructure/project/ProjectPreviewLookupImplementation.java

```diff
@@ -0,0 +1,123 @@
+package life.qbic.projectmanagement.infrastructure.project;
+
+import java.util.Collection;
+import java.util.List;
+import java.util.Objects;
+import life.qbic.application.commons.OffsetBasedRequest;
+import life.qbic.application.commons.SortOrder;
+import life.qbic.projectmanagement.application.ProjectPreview;
+import life.qbic.projectmanagement.application.api.ProjectPreviewLookup;
+import life.qbic.projectmanagement.domain.model.project.ProjectId;
+import org.springframework.context.annotation.Scope;
+import org.springframework.data.domain.Sort;
+import org.springframework.data.domain.Sort.Order;
+import org.springframework.data.jpa.domain.Specification;
+import org.springframework.stereotype.Component;
+
+/**
+ * Basic implementation to query project preview information
+ *
+ */
+@Component
+@Scope("singleton")
+public class ProjectPreviewLookupImplementation implements ProjectPreviewLookup {
+
+  private final ProjectPreviewRepository projectPreviewRepository;
+
+  public ProjectPreviewLookupImplementation(ProjectPreviewRepository projectPreviewRepository) {
+    Objects.requireNonNull(projectPreviewRepository);
+    this.projectPreviewRepository = projectPreviewRepository;
+  }
+
+  @Override
+  public List<ProjectPreview> query(int offset, int limit) {
+    return projectPreviewRepository.findAll(new OffsetBasedRequest(offset, limit)).getContent();
+  }
+
+  @Override
+  public List<ProjectPreview> query(String filter, int offset, int limit,
+      List<SortOrder> sortOrders, Collection<ProjectId> projectIds) {
```

Comment:
NIT: Have you tried this using PreFilter annotation with the projectIds? It sounds easier to me, so you don't have to manually filter.